sshd-core/src/test/java/org/apache/sshd/common/forward/ConcurrentConnectionTest.java

```diff
@@ -0,0 +1,257 @@
+            LOG.error("Error", t);
+        }
+    }
+
+    private void serverSocketLoop(AtomicInteger activeServers, Socket s) {
+        try {
+            LOG.debug("Active Servers: {}", activeServers.incrementAndGet());
+            final byte[] buf = new byte[PAYLOAD_TO_SERVER.length];
+            final long r = s.getInputStream().read(buf);
+            LOG.debug("Read {} payload from client", r);
+            s.getOutputStream().write(PAYLOAD_TO_CLIENT);
+            LOG.debug("Wrote payload to client");
+            s.close();
+            LOG.debug("Active Servers: {}", activeServers.decrementAndGet());
+        } catch (final Throwable t) {
+            LOG.error("Error", t);
+        }
+    }
+
+    @After
+    public void stopTestServer() throws Exception {
+        testServerSock.close();
+        testServerThread.interrupt();
+    }
+
+    @BeforeClass
+    public static void startSshServer() throws IOException {
+        LOG.debug("Starting SSHD...");
+        server = SshServer.setUpDefaultServer();
+        server.setPasswordAuthenticator((u, p, s) -> true);
+        server.setKeyPairProvider(new SimpleGeneratorHostKeyProvider());
+        server.setNioWorkers(SSHD_NIO_WORKERS);
+        server.setForwardingFilter(AcceptAllForwardingFilter.INSTANCE);
+        server.start();
+        sshServerPort = server.getPort();
+        LOG.debug("SSHD Running on port {}", server.getPort());
+    }
+
+    @AfterClass
+    public static void stopServer() throws IOException {
+        if (!server.close(true).await(TIMEOUT)) {
+            LOG.warn("Failed to close server within {} sec.", TimeUnit.MILLISECONDS.toSeconds(TIMEOUT));
+        }
+    }
+
+    @Before
+    public void createClient() throws IOException {
+        final SshClient client = SshClient.setUpDefaultClient();
+        client.setForwardingFilter(AcceptAllForwardingFilter.INSTANCE);
+        client.start();
+        LOG.debug("Connecting...");
+        session = client.connect("user", TEST_LOCALHOST, sshServerPort).verify(TIMEOUT).getSession();
+        LOG.debug("Authenticating...");
+        session.addPasswordIdentity("foo");
+        session.auth().verify(TIMEOUT);
+        LOG.debug("Authenticated");
+    }
+
+    @After
+    public void stopClient() throws Exception {
+        LOG.debug("Disconnecting Client");
+        try {
+            assertTrue("Failed to close session", session.close(true).await(TIMEOUT));
+        } finally {
+            session = null;
+        }
+    }
+
+    @Test
+    /*
+     * Run PORT_FORWARD_CLIENT_COUNT simultaneous server threads.
+     * 
+     * Emulates a web browser making a number of simultaneous requests on
+     * different connections to the same server HTTP specifies no more than two,
+     * but most modern browsers do 6 or more.
+     */
+    public void testConcurrentConnectionsToPortForward() throws Exception {
+        final SshdSocketAddress remote = new SshdSocketAddress(TEST_LOCALHOST, 0);
```

Comment:
I think it is easier to read if documentation precedes the annotation (i.e., annotation is attached to the method). BTW, is _checkstyle_ allowing this ?